qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -61,7 +63,18 @@ def __init__(self, basis_gates=None, target=None):
             self.basis = set(target.operation_names)
 
         self._var_z_map = {"rz": RZGate, "p": PhaseGate, "u1": U1Gate}
-        self.requires.append(CommutationAnalysis())
+
+        self._z_rotations = ["p", "z", "u1", "rz", "t", "s"]
+        self._x_rotations = ["x", "rx"]  # sx too?
```

Comment:
We could cancel 4 sx's I guess, but does the pass handle that?

Reply:
Not yet, but it would be easy to add. I'll defer that to a follow-up to keep changes minimal for 1.2.